lib/dry/validation/schema.rb

```diff
@@ -92,6 +94,14 @@ def [](name)
       def predicates
         self.class.predicates
       end
+
+      private
+
+      def validate_input(input)
+        unless input.is_a?(Hash)
+          raise ArgumentError, "expected a Hash but received +#{input.inspect}+"
+        end
```

Comment:
@SunnyMagadan I agree, that's why I said the name wasn't very good. I was just trying to describe something that would validate the given input would respond correctly to the `attr` preconditions, just like the `hash?` might check that the input is a hash or responds to the needed methods.